### PR TITLE
[To rel/1.0][IOTDB-5253] Fix chunkMetadata mem metric

### PR DESCRIPTION
**Pull request by SpriCoder**
https://issues.apache.org/jira/browse/IOTDB-5253

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessorInfoMetrics.java [hunk 1]
```diff
@@ -64,11 +65,12 @@ public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
     TsFileProcessorInfoMetrics that = (TsFileProcessorInfoMetrics) o;
-    return memCost == that.memCost && Objects.equals(storageGroupName, that.storageGroupName);
+    return Objects.equals(storageGroupName, that.storageGroupName)
```
Comment (OneSizeFitsQuorum) [inline server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessorInfoMetrics.java:68]: Do we need to override the equal and hashcode of TsFileProcessorInfo?

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessorInfoMetrics.java [hunk 2]
```diff
@@ -64,11 +65,12 @@ public boolean equals(Object o) {
...
+        && Objects.equals(tsFileProcessorInfo, that.tsFileProcessorInfo);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(storageGroupName, memCost);
+    return Objects.hash(storageGroupName, tsFileProcessorInfo);
```
Comment (OneSizeFitsQuorum) [inline server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessorInfoMetrics.java:74]: same

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessorInfoMetrics.java (hunk 2 above)
Comment (SpriCoder) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessorInfoMetrics.java]: fixed~

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessorInfoMetrics.java (hunk 1 above)
Comment (SpriCoder) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessorInfoMetrics.java]: fixed~

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**